admin_panel/src/commonMain/kotlin/net/thechance/mena/admin_panel/presentation/component/ClickableTextRow.kt

```diff
@@ -0,0 +1,64 @@
+package net.thechance.mena.admin_panel.presentation.component
+
+import androidx.compose.animation.animateColorAsState
+import androidx.compose.animation.core.tween
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.size
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.unit.dp
+import net.thechance.mena.designsystem.presentation.component.icon.Icon
+import net.thechance.mena.designsystem.presentation.component.text.Text
+import net.thechance.mena.designsystem.presentation.theme.theme.Theme
+
+@Composable
+fun ClickableTextRow(
+    text: String,
+    icon: Painter,
+    modifier: Modifier = Modifier,
+    isSelected: Boolean = false,
+    onClick: () -> Unit = {}
+) {
+    val tint by animateColorAsState(
+        targetValue = if (isSelected) {
+            Theme.colorScheme.success
+
+        } else {
+            Theme.colorScheme.shadePrimary
+        },
+        animationSpec = tween(durationMillis = 300),
+        label = "iconTint"
+    )
```

Comment:
We can move the if condition outside for easier readability
```suggestion
    val iconTent = if (isSelected) {
            Theme.colorScheme.success
        } else {
            Theme.colorScheme.shadePrimary
        }
        
    val animatedIconTent by animateColorAsState(
        targetValue = iconTent,
        animationSpec = tween(durationMillis = 300),
        label = "iconTint"
    )
```